cmake/macros/macro_expand_instantiations.cmake

```diff
@@ -78,7 +78,12 @@ MACRO(EXPAND_INSTANTIATIONS _target _inst_in_files)
   #
   FOREACH(_build ${DEAL_II_BUILD_TYPES})
     STRING(TOLOWER ${_build} _build_lowercase)
+
     ADD_DEPENDENCIES(${_target}.${_build_lowercase} ${_target}.inst)
```

Comment:
CMake distinguishes between:

 - Build time dependencies have to be made explicit via explicitly stating top level target dependencies (aka `ADD_DEPENDENCIES( ... )`). This is for ordering targets within makefiles.

 - CMake also has a mechanism for creating include dependencies (on internal, or external files). This information is used to determine whether a target needs to be recompiled.

CMake dropped implicit ordering of targets with information from include dependencies a long time ago in order to support different generators.

As a matter of fact, CMake correctly creates a set of include dependencies for the .cu files.